### DISCUSSION TITLE
PorousFlow - Excavation construction process

### DISCUSSION THREAD

**Discussion by fcastiloo** [idaholab/moose#20601] [Q&A Meshing] [answered]
Hi MOOSE,
I am new in the MOOSE world. I am simulating an excavation process with PorousFlow module. I think I understand how it works and the coupling among physics. I also studied the simulation of a coal mine where the excavated rock gets permeability zero.

But I was wondering if it is possible to update the mesh and somehow remove the nodes where the excavation occurs, hence, I can also update the boundary conditions after the excavation. I know I can pass information between meshes with MultyApps, but I could not find a good example. Also, I would like to have a unique output file where I can see how the mesh changes over time. Please let me knwo if this doesnt make sense.

Thanks and regards,
Felipe

**Comment by WilkAndy** [comment]
I would like others to comment on this, but one way is to use `BlockDeletionGenerator`, https://mooseframework.inl.gov/source/meshgenerators/BlockDeletionGenerator.html .   Please note that your simulation might "go crazy" when you suddenly delete a block of elements, and you might have to soften the material before deleting it.

**Reply by fcastiloo** [reply]
Hi,
Thanks! The only problem is that with `BlockDeletionGenerator`, every time the mesh changes, I will have to make a new MOOSE input file.
Maybe someone else has another idea. Usually, in this kind of problems, the  simulation software allow the "deactivation" of the nodes. But my experience is only as user.

**Comment by GiudGiud** [comment, marked as answer]
Hello

So you'll want to use a https://mooseframework.inl.gov/source/userobject/CoupledVarThresholdElementSubdomainModifier.html
This should be able to change the subdomain IDs of elements, and you can restrict them out of the domain of application of kernels that way, and it should also be able to update the sidesets for applying boundary conditions.

One restriction is that you have to frame the conditions for the mesh modification as a variable threshold. Hope that works for you

Guillaume

**Reply by fcastiloo** [reply]
Hi GuidGuid,
Thanks!! This looks like what I need. One naive question: How can I define a rectangule as a function in MOOSE?

**Reply by GiudGiud** [reply]
What do you mean a rectangle? This doesnt take a rectangle, this takes a single real
If this is for something else, you can make a 'rectangle' (function is 0 outside a rectangle) with both of those:
https://mooseframework.inl.gov/source/functions/PiecewiseMulticonstant.html
https://mooseframework.inl.gov/source/functions/MooseParsedFunction.html

**Reply by fcastiloo** [reply]
Thanks! In the example (https://mooseframework.inl.gov/source/userobject/CoupledVarThresholdElementSubdomainModifier.html), in the overview section, a function of a circle is defined and then re-defined as an auxvar. I was wondering how to define a rectangule instead of a circle. But it looks like I can do that with PasedFunction. 
Thanks again!
